core/src/main/java/hudson/security/PasswordHashEncoder.java

```diff
@@ -0,0 +1,7 @@
+package hudson.security;
+
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+public interface PasswordHashEncoder extends PasswordEncoder {
```

Comment:
does the interface need to be `public`?  can we keep it package scoped so that it does not leak and can be more easily changed later if needed

Reply:
yes changed